### PR TITLE
feat(headerMenu): add an option to show/hide column menu buttons

### PR DESCRIPTION
**Pull request by classix-od**
Added an option to show/hide column menu buttons. The option can be set for specific columns or in general.

### CODE REVIEW

File: src/js/core/directives/ui-grid-column-menu.js [hunk 1]
```diff
@@ -401,8 +401,11 @@ function ($timeout, gridUtil, uiGridConstants, uiGridColumnMenuService, $documen
           $scope.hideMenu( true );
 
           if ($scope.col) {
-            //Focus on the menu button
-            gridUtil.focus.bySelector($document, '.ui-grid-header-cell.' + $scope.col.getColClass()+ ' .ui-grid-column-menu-button', $scope.col.grid, false);
+            //Focus on the menu button if this is shown, otherwise focus the column header
+            var elementToFocus = '.ui-grid-header-cell.' + $scope.col.getColClass();
+            elementToFocus += ($scope.grid.options.showColumnMenuButtons && $scope.col.colDef.showColumnMenuButton !== false)?
```
Comment (mportuga) [inline src/js/core/directives/ui-grid-column-menu.js]: This is redundant. if $scope.grid.options.showColumnMenuButton returns truthy, then it is obviously different than false. You only need one of these checks.

File: src/templates/ui-grid/uiGridHeaderCell.html [hunk 2]
```diff
@@ -37,7 +37,7 @@
     tabindex="0"
     ui-grid-one-bind-id-grid="col.uid + '-menu-button'"
     class="ui-grid-column-menu-button"
-    ng-if="grid.options.enableColumnMenus && !col.isRowHeader  && col.colDef.enableColumnMenu !== false"
+    ng-if="grid.options.enableColumnMenus && grid.options.showColumnMenuButtons && !col.isRowHeader  && col.colDef.enableColumnMenu !== false && col.colDef.showColumnMenuButton !== false"
```
Comment (mportuga) [inline src/templates/ui-grid/uiGridHeaderCell.html]: Same comment as above

File: src/js/core/directives/ui-grid-column-menu.js (hunk 1 above)
Comment (classix-od) [reply src/js/core/directives/ui-grid-column-menu.js]: They are not the same. Please look carefully, the first is general and the second is column specific.

File: src/templates/ui-grid/uiGridHeaderCell.html (hunk 2 above)
Comment (classix-od) [reply src/templates/ui-grid/uiGridHeaderCell.html]: Same comment as above

**Comment by mportuga** [comment]
Closing due to age and conflicts.

**Pull request closed without merging**